modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -12,6 +12,14 @@
 #include <numeric>
 namespace opencv_test { namespace {
 
+struct FilteredDetections{
+    std::vector<int> keep_classIds;
+    std::vector<float> keep_confidences;
+    std::vector<Rect2d> keep_boxes;
+};
```

Comment:
I was thinking constructing it that way, but then opted to proceed with current version because this way I thought it is more self contained (a part from needing to have struct). If you think it is better to return by argument, I will change it.